pycbc/psd/variation.py

```diff
@@ -210,10 +210,187 @@ def find_trigger_value(psd_var, idx, start, sample_rate):
+    adjacent elements in the array. This array is then further averaged every
+    second to produce the PSD variation timeseries.
+
+    Parameters
+    ----------
+    strain : pycbc.timeseries
+        Live data being searched through by the PyCBC Live search.
+    full_filt : numpy.ndarray
+        A filter created by `live_create_filter`.
+    increment : float
+        The number of seconds in each increment in the PyCBC Live search.
+    data_trim : float
+        The number of seconds to be trimmed from either end of the convolved
+        timeseries to prevent artefacts.
+    short_stride : float
+        The number of seconds to average the PSD variation timeseries over to
+        remove the effects of short duration glitches.
+
+    Returns
+    -------
+    psd_var : pycbc.timeseries
+        A timeseries containing the PSD variation values.
+
+    """
+    sample_rate = int(strain.sample_rate)
+
+    # Grab the last increments worth of data, plus padding for edge effects.
+    astrain = strain.time_slice(strain.end_time - increment - (data_trim * 3),
+                                strain.end_time)
+
+    # Convolve the data and the filter to produce the PSD variation timeseries,
+    #  then trim the beginning and end of the data to prevent edge effects.
+    wstrain = sig.fftconvolve(astrain, full_filt, mode='same')
+    wstrain = wstrain[int(data_trim * sample_rate):-int(data_trim * sample_rate)]
+
+    # Create a PSD variation array by taking the mean square of the PSD
+    #  variation timeseries every short_stride
+    short_ms = numpy.mean(
+        wstrain.reshape(-1, int(sample_rate * short_stride)) ** 2, axis=1)
+
+    # Define an array of averages that is used to substitute outliers
+    #  variation array
+    ave = 0.5 * (short_ms[2:] + short_ms[:-2])
+    outliers = short_ms[1:-1] > (2. * ave)
+    short_ms[1:-1][outliers] = ave[outliers]
+
+    # Calculate the PSD variation every second by averaging over short_ms every
+    #  samples_per_second.
+    m_s = []
+    samples_per_second = 1 / short_stride
+    for idx in range(int(len(short_ms) / samples_per_second)):
+        start = int(samples_per_second * idx)
+        end = int(samples_per_second * (idx + 1))
+        m_s.append(numpy.mean(short_ms[start:end]))
+
+
+    m_s = numpy.array(m_s, dtype=wstrain.dtype)
+
+    psd_var = TimeSeries(m_s,
+                         delta_t=1.0,
+                         epoch=strain.end_time - increment - (data_trim * 2))
+
+    return psd_var
+
+
+def live_find_var_value(triggers,
+                        psd_var_timeseries):
+    """
+    Extract the PSD variation values at trigger times by linear interpolation.
+
+    Parameters
+    ----------
+    triggers : dict
+        Dictionary containing input trigger times.
+    psd_var_timeseries : pycbc.timeseries
+        A timeseries containing the PSD variation value for each second of the
+        latest increment in PyCBC Live. Created by live_calc_psd_variation.
```

Comment:
Where the timeseries came from is not relevant to understanding the function.
```suggestion
        latest increment in PyCBC Live.
```